### PR TITLE
Adding custom path for command line pool configs and Improved command line UI

### PR DESCRIPTION
**Pull request by eljhfx**
PR for issue #48

- Created a test config file in `crates/cli/config-test` to make sure it was working
- Changed the `struct Cli` to `struct Args to be consistent with the clap documentation
- Changed the `config` field for `enum Commands` from a `bool` to a `String` with a default value `""`
- Added logic to handle the case where a path is given as part of the `config` command such that the proper config file is used (that follows the path given in the command line)
- Reformatted the terminal output to look nicer and give users more, useful info that gives them confidence their commands were parsed correctly

### CODE REVIEW

File: crates/cli/config-test/config.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+rpc_url = "https://eth-mainnet.g.alchemy.com/v2/I93POQk49QE9O-NuOz7nj7sbiluW76it"
```
Comment (ts0yu) [inline crates/cli/config-test/config.toml:1]: this file seems redundant as we already have an example in the config folder?

File: crates/cli/src/main.rs [hunk 2]
```diff
@@ -36,9 +36,9 @@ enum Commands {
         #[arg(default_value = "5")]
         bp: String,
 
-        /// Set this flag to use the config.toml.
-        #[arg(short = 'c', long = "config", action)]
-        config: bool,
+        /// Set this flag to use a config.toml.
+        #[arg(short, long, action, default_value = "")]
```
Comment (ts0yu) [inline crates/cli/src/main.rs:40]: it'd be nice to allow the user to just pass the `--config` flag and be able to use the default paths

File: crates/cli/src/config/mod.rs [hunk 3]
```diff
@@ -53,20 +53,32 @@ pub struct Config {
...
     /// Public constructor function to instantiate a representation of a config file.
-    pub fn new() -> Result<Self, ConfigError> {
+    pub fn new(command_path: &String) -> Result<Self, ConfigError> {
         let mut content = String::new();
 
-        let config_filepaths: [&str; 2] = [
-            "./crates/cli/src/config.toml",
-            "./crates/cli/src/Config.toml",
-        ];
+        match command_path == "" {
```
Comment (ts0yu) [inline crates/cli/src/config/mod.rs:59]: doesn't seem to me there is any circumstance in which it is called with an empty string

**Review by ts0yu — CHANGES_REQUESTED** [review]

**Comment by ts0yu** [comment]
need to get lint passing too

File: crates/cli/src/config/mod.rs [hunk 4]
```diff
@@ -53,20 +53,32 @@ pub struct Config {
 
 impl Config {
     /// Public constructor function to instantiate a representation of a config file.
-    pub fn new() -> Result<Self, ConfigError> {
+    pub fn new(command_path: &String) -> Result<Self, ConfigError> {
```
[the remark this answers is not in the document]
Comment (0xJepsen) [reply crates/cli/src/config/mod.rs:56]: I think it should be.

File: crates/cli/src/config/mod.rs (hunk 3 above)
Comment (0xJepsen) [reply crates/cli/src/config/mod.rs:59]: Would also be taken care of it the command is an option. Still leveraging the default paths

**Review by 0xJepsen — CHANGES_REQUESTED** [review]
I think over all the implementation looks really good. I think the main thing is that we want to retain the old functionality and it the default behavior, i think making the arg an option is the right approach for this.

File: crates/cli/src/config/mod.rs (hunk 3 above)
Comment (eljhfx) [reply crates/cli/src/config/mod.rs:59]: The `--config` argument is a string, the default is value is `""` (meaning if they don't input a path it's an empty string) and iterates through the default values.

File: crates/cli/src/main.rs (hunk 2 above)
Comment (eljhfx) [reply crates/cli/src/main.rs:40]: Ahhh will fix.

File: crates/cli/config-test/config.toml (hunk 1 above)
Comment (eljhfx) [reply crates/cli/config-test/config.toml:1]: It's a test case to check that you can set a non-default path. Can remove it if you don't think it's useful to have the test case here though.

File: crates/cli/src/config/mod.rs (hunk 3 above)
Comment (eljhfx) [reply crates/cli/src/config/mod.rs:59]: Yeah was under the impression `default_value = ""` meant if you don't specify a path it won't have an error, but see this is not the case. Will fix

**Comment by eljhfx** [comment]
Made the changes and ready for re-review @ts0yu

**Review by 0xJepsen — APPROVED** [review]
looks good to me!

File: crates/cli/config-test/config.toml (hunk 1 above)
Comment (ts0yu) [reply crates/cli/config-test/config.toml:1]: oh I see, that makes sense. I don't really mind either way

**Review by ts0yu — APPROVED** [review]
sweet - really good stuff.

**Pull request merged**